sdk/storage/Azure.Storage.Files.Shares/src/ShareDirectoryClient.cs

```diff
@@ -1876,9 +2043,41 @@ await GetSubdirectoryClient(subdirectoryName)
         [ForwardsClientCalls]
         public virtual async Task<Response> DeleteFileAsync(
             string fileName,
+            FileRequestConditions conditions = default,
             CancellationToken cancellationToken = default) =>
             await GetFileClient(fileName)
-                .DeleteAsync(cancellationToken)
+                .DeleteAsync(
+                    conditions,
+                    cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="DeleteFileAsync(string, CancellationToken)"/>
+        /// operation immediately removesthe file from the storage account.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/delete-file2"/>.
+        /// </summary>
+        /// <param name="fileName">The name of the file.</param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> on successfully deleting.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will bse thrown if
+        /// a failure occurs.
+        /// </remarks>
+        [ForwardsClientCalls]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0002:DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.", Justification = "<Pending>")]
```

Comment:
nit: Suggest using the pragma style suppressions for the sake of consistency.

Reply:
fixed.